src/server.ts

```diff
@@ -15,17 +15,27 @@ router.get('/rovers', async (req:any, res:any) => {
     res.send(roversList);
 });
 
-router.get('/rovers/photos', async (req:any, res:any) => {
-    const roverPhotoList = await requestHelper.getRoverPhotos();
-    let roverPhotoImages: string [] = [];
-    roverPhotoList.photos.map((photo) => {
-        roverPhotoImages.push(photo.img_src);
-    })
-    console.log(roverPhotoImages);
-    roverPhotoImages.forEach((photo) => {
+router.get('/rovers/:roverName/photos/:cameraType', async (req:any, res:any) => {
 
-    })
-    res.send(roverPhotoImages);
+    let roverPhotoList;
+    try {
+        roverPhotoList = await requestHelper.getRoverPhotos(req.params.roverName, req.params.cameraType);
+    } catch (error) {
+        console.log(error);
+        res.sendStatus(400);
+    } finally {
+        if (roverPhotoList != null) {
+            let roverPhotoImages: string [] = [];
+            roverPhotoList.photos.map((photo) => {
+                roverPhotoImages.push(photo.img_src);
+            })
+            console.log(roverPhotoImages);
+            roverPhotoImages.forEach((photo) => {
+
+            })
+            res.send(roverPhotoImages);
+        }
+    }

```

Comment:
```suggestion
        return res.sendStatus(400);
    }
    
    if (roverPhotoList != null) {
        let roverPhotoImages: string [] = [];
        roverPhotoList.photos.map((photo) => {
            roverPhotoImages.push(photo.img_src);
        })
        console.log(roverPhotoImages);
        roverPhotoImages.forEach((photo) => {})
        
        res.send(roverPhotoImages);
    }
}
```

This is express-specific but... when you're dealing with responses, you can just `return` the response i.e. `return res.sendStatus(400);`

In which case, any thing after the `try` `catch` blocks only have to consider the happy path, so we can remove the `finally` here.

One thing to consider though - what if the NASA API request succeeds (so won't be caught), but `roverPhotoList` is null? First of all, is that possible? If so we currently don't give any response when that happens (the website will hang). If it isn't possible, we can remove the null check `if (roverPhotoList != null)`.